src/Tests/InstrEngineTests/InstrEngineTests/TestAppAssembler.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InstrEngineTests
+{
+    internal class TestAppAssembler
+    {
+        internal static bool AssembleFile(IEmbeddedResourceFile embeddedResource, string directoryPath, string assemblyName, bool isDebug, bool is64Bit, bool isExe = false)
+        {
+            ProcessStartInfo psi = new ProcessStartInfo();
+            IEmbeddedResourceFile ilasm = EmbeddedResourceUtils.GetTestResourceFile("ilasm.exe");
+            if (ilasm == null)
+            {
+                return false;
+            }
+
+            psi.FileName = ilasm.FilePath;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardOutput = true;
+            psi.UseShellExecute = false;
+            string debugSwitch = isDebug ? "/DEBUG=OPT" : string.Empty;
+            string bitSwitch = is64Bit ? "/X64" : "/32BITPREFERRED";
+            string outputType = isExe ? "/EXE" : "/DLL";
```

Comment:
I'm assuming this behavior is intentional for our specific use-case, but figured I'd point it out - this is the opposite of what you would expect when using `ilasm` - the default is to assume `/EXE` unless `/DLL` is specified, whereas with the default for `isExe` set to false, we automatically default to `/DLL`. Assuming this is the behavior we want by default, perhaps a comment would be useful to call this out.

Reply:
This is by design because it follows the same convension as compiling CSharp. I think with CSharp, we are building exes all the time. We may or may not follow that same decision when we build the assemblies from IL.